### ISSUE TITLE
GDScript 2.0: Inheritance is not working for return type for method overriding

### ISSUE THREAD

**Issue by Janglee123** [godotengine/godot#72403]
### Godot version

v4.0.beta16.official [518b9e580]

### System information

Linux

### Issue description

The following code supposed to work as `BReturn` extends `AReturn`. But I get error that `ClassB.foo`'s function signature doesn't match with the parent and expects only `AReturn` as return type. 

```
class AReturn:
	pass

class BReturn extends AReturn:
	pass

class ClassA:
	func foo() -> AReturn:
		return AReturn.new()

class ClassB extends ClassA:
	func foo() -> BReturn:
		return BReturn.new()
```

### Steps to reproduce

Try out the above code

### Minimal reproduction project

N/A

**Comment by vmedea** [comment]
I've also noticed this, but wasn't sure "covariant return types" like this are intended to be supported in GDscript (if not it's more of a feature request). It's not universally supported in object oriented languages.

**Comment by anvilfolk** [comment]
Yeah, IIRC the method's signatures must match for methods with the same name :)

**Comment by HolonProduction** [comment]
Can't reproduce in v4.7.1.stable.flathub [a13da4feb]
